Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,197 @@
+using Content.Server.Botany.Components;
+using Content.Server.PowerCell;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction;
+using Content.Shared.PlantAnalyzer;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using System.Linq;
+using System.Text;
+
+namespace Content.Server.Botany.Systems;
+
+public sealed class PlantAnalyzerSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PowerCellSystem _cell = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PlantAnalyzerComponent, AfterInteractEvent>(OnAfterInteract);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerSetMode>(OnModeSelected);
+    }
+
+    private void OnAfterInteract(Entity<PlantAnalyzerComponent> ent, ref AfterInteractEvent args)
+    {
+        if (args.Target == null || !args.CanReach || !HasComp<SeedComponent>(args.Target) && !HasComp<PlantHolderComponent>(args.Target) || !_cell.HasActivatableCharge(ent, user: args.User))
+            return;
+
+        if (ent.Comp.DoAfter != null)
+            return;
+
+        if (ent.Comp.Settings.AdvancedScan)
+        {
+            var doAfterArgs = new DoAfterArgs(EntityManager, args.User, ent.Comp.Settings.AdvScanDelay, new PlantAnalyzerDoAfterEvent(), ent, target: args.Target, used: ent)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                NeedHand = true
+            };
+            _doAfterSystem.TryStartDoAfter(doAfterArgs, out ent.Comp.DoAfter);
```

Comment:
I think I said this before in an earlier review?

Starting doafter should really be done in shared, or it will feel unresponsive since you won't see the doafter until after the server informs the client that the doafter should be started.

`OnDoAfter` callback can be empty on the client system and only implemented on the server.